### DISCUSSION TITLE
Double-difference interferogram

### DISCUSSION THREAD

**Discussion by riyasmjgeo** [isce-framework/isce2#513] [General]
Dear All,

I want to create a double-difference interferogram of an ice shelf for identifying the grounding line. However, I have doubts regarding the flat-earth correction and topo-phase removal procedures. I am requesting all to kindly provide your feedback on the questions below.  
1. What happens if I don't do those corrections? The effects of both flat-earth and topo-phase will be present in both the interferograms that are used for double differencing. Hence, will it cancel out while subtracting?
2. If I need to do the corrections, should I do them on both the interferograms and find the difference? Or should I do the corrections in the final interferogram (difference between both interferograms) only?
3. The steps involved in topsApp.py are ['startup', 'preprocess', 'computeBaselines', 'verifyDEM', 'topo']
    ['subsetoverlaps', 'coarseoffsets', 'coarseresamp', 'overlapifg', 'prepesd']
    ['esd', 'rangecoreg', 'fineoffsets', 'fineresamp', 'ion']
    ['burstifg', 'mergebursts', 'filter', 'unwrap', 'unwrap2stage']
    ['geocode', 'denseoffsets', 'filteroffsets', 'geocodeoffsets']
    In which step the corrections are carried out? Is it in the step 'burstifg'? If so, is there any option to get the interferogram without having those corrections? 
4. Most importantly, the theory is that the effect of Ice flow in phase difference will be canceled out if the ice velocity is the same during the periods between both InSAR pairs. However, the ice flow velocity in the shelf regions will be mostly >1m/day. In the case of Sentinel-1, the minimum 6-day interval can produce 6 meters of movement. If we consider a 0.1% change in the ice velocity, it will cause a considerable difference (>20%) in the phase difference between the interferograms. In that case, is this method valid to use for avoiding the effects of ice flow? 

I am aware that most questions are not related to ISCE, but I don't find any other forum with these many experts in it. So, kindly help in this regard.  

Thank you

**Comment by fdelgadodelapuente** [comment]
Hi Riyas,

ISCE works in a very different way than the theory described in almost all of the InSAR books and review articles which applies to the old ROI_PAC workflows. The flat-earth correction and topo-phase removal procedures are carried out in a single step in ISCE. In stripmapApp and topsApp the range.off file is the range offset from the forward geometric coregistration required for resampling secondary to reference, and also the flat-earth correction. The SLCs that you use to calculate the interferogram include this correction. 

The differential interferogram is calculated with 
`diff_int = A*conj(B)*exp(-J*4*PI*R*C/W)  `

with A the reference SLC, B the secondary SLC, C the range offset file, J the imaginary constant, R the slant range pixel spacing and W the radar wavelength. So you can reverse this formula with imageMath.py to get an interferogram without these corrections 

`uncorrected_int = A*conj(B)*exp(J*4*PI*R*C/W)`

Francisco

**Reply by riyasmjgeo** [reply]
@fdelgadodelapuente Thank you for the information. It is exciting and shocking since every material I read states that the corrections are done on the generated interferogram. Is this the case in ISCE only or in other applications as well? 
In the ESA sentinel-1 tutorial also, it is mentioned to perform the corrections in the generated interferogram. But, what you stated makes sense, and I think it is effective. Thanks again

**Reply by fdelgadodelapuente** [reply]
Pretty much all the books and review articles describe these corrections in the old way -- first flat earth and then DEM.  In ROI_PAC these corrections were carried out each independent from each other by the module diffnsim. I don't know how they are implemented in other software like GAMMA, GMTSAR, DORIS, SNAP, etc.

You could test them with the following code for ENVISAT data with stripmapApp (4 looks in range, 20 in azimuth). After the interferogram step run the following

`
imageMath.py -e='a*conj(b)' --a=reference_slc/reference.slc  --b=coregisteredSlc/refined_coreg.slc -o d.int -t CFLOAT
`

`looks.py -i d.int -a 20 -r 4`

`mdx.py d.20alks_4rlks.int interferogram/topophase.flat`

**Reply by riyasmjgeo** [reply]
Thank you @fdelgadodelapuente

**Comment by piyushrpt** [comment]
You can set up a zero height DEM and just run the topo and geo2rdr steps and the range.off then would represent the flat earth phase component. However, do not use this for coregistration as this would not take topography into account. The radar can only image what it sees and does not know distinction between flat earth and topography. Removing flat earth component was done in early days when DEMs were not of great quality and also to reduce fringe rate for unwrapping if you were using InSAR itself to generate the DEM.

**Reply by riyasmjgeo** [reply]
Interesting! Thank you for the information. I have another doubt I intended to ask you @piyushrpt. 
I believe the atmospheric bulk delay is corrected in the range coregistration step (I could be wrong!). If so, are we removing the atmospheric delay a second time while using a weather model for generating a time-series, such as in GIAnT?

**Reply by fdelgadodelapuente** [reply]
Atmospheric phase delays are not corrected in ISCE. You need to use external software like Pyaps in MintPy or TRAIN to try to correct them.  And I mean to try because these corrections never work perfectly for all of the interferograms in a data set. They also depend a lot on the type of noise statistics (turbulence, topo correlated, etc) and the place on Earth you are

**Reply by riyasmjgeo** [reply]
I got confused regarding this atmospheric correction on watching a tutorial by Piyush (20200811 UNAVCO-TopsApp, video time- 2:47:00). I guess I am mistaken. Thanks for the clarification @fdelgadodelapuente

**Reply by piyushrpt** [reply]
InSAR is a relative measurement. Applying a constant bulk shift does not change the relative phase shifts between pixels.
